src/shape_inference.jl

```diff
@@ -379,6 +379,27 @@ for func in ["Sum", "Prod", "Min", "Max", "All", "Any", "Mean"]
     end
 end
 
+for func in ["ArgMax", "ArgMin"]
+    register_shape(func) do op
```

Comment:
This could be written 
`register_shape.( ["ArgMax", "ArgMin"]) do`

rather than a for loop.
But that might be being too clever.

Reply:
I like that. Since I copied most of this block from the block above I'll introduce the same change there as well.

Reply:
I agree